addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.DiscussSidebarCategory" t-inherit-mode="extension">
+        <xpath expr="//*[@t-ref='categoryHeader']" position="after">
+            <div t-if="state.editing === category.id" class="p-2" t-ref="selector">
+                <ChannelSelector category="category" onValidate.bind="stopEditing" autofocus="true" close.bind="stopEditing" />
+            </div>
+        </xpath>
+        <xpath expr="//*[@t-ref='openCategoryBtn']" position="after">
```

Comment:
Nitpicking: prefer `name` when only used in xpath

Reply:
If you really don't want ref, the 2nd best way is to use data- attribute.

Reply:
I don't like polluting the DOM for XPath purposes either.

Reply:
No one really care for `name` polluting the DOM. `t-ref` is fine but this introduces some OWL internal processing, which could causes subtle issues.

`name` attribute is much simpler and to the point, hence why I'd prefer it. But if you insist with `t-ref` then fine, both should work the same so it doesn't really matter.

Reply:
Indeed, not a big deal. I applied your suggestion.